ui/src/api.ts

```diff
@@ -119,13 +119,14 @@ const delay = (delay: number) => {
 
 export const withRetry = async <T>(
   request: () => Promise<T>,
-  delayTime: number
+  delayTime: number,
+  retries: number
 ): Promise<T> => {
-  for (let retries = 0; ; retries++) {
+  for (; ; retries--) {
     try {
       return await request();
     } catch (error) {
-      if (error.message !== "Failed to fetch" || retries > 200) {
+      if (error.message !== "Failed to fetch" || retries > 0) {
```

Comment:
```suggestion
      if (error.message !== "Failed to fetch" || retries < 0) {
```

Or am I understanding the loop wrong?

Reply:
Great catch!